conda_build/metadata.py

```diff
@@ -46,18 +55,12 @@
 )
 
 if TYPE_CHECKING:
+    from pathlib import Path
     from typing import Any, Literal, Self
 
     OutputDict = dict[str, Any]
     OutputTuple = tuple[OutputDict, "MetaData"]
 
-try:
-    import yaml
-except ImportError:
-    sys.exit(
-        "Error: could not import yaml (required to read meta.yaml "
-        "files of conda recipes)"
-    )
```

Comment:
Was that basically an implicit dependency? Do we require it now in meta.yml?

Reply:
I'm surprised that it doesn't say "ruamel.yaml" but that package is certainly a regular dependency now.

Reply:
It was previously an optional dependency but is currently [listed as a requirement in `meta.yml`](https://github.com/conda/conda-build/commit/f60aa9a3bee04e0d6d39166dba231786c3b14939#diff-0286c1d624d881984aef773a82433d955692c2fcbc773e7a6230fdb9602a23a3R31).

This particular code change is more "cleanup" vs "remove `sys.exit`" (per work originally done in [this PR](https://github.com/conda/conda-build/pull/5255/files#diff-a3f5613bda9366e31149c65731327047d4e29204c9e0508fd55416115b39a6bdL50-L57)), so I will revert this change; once the broken-out parts of the "remove `sys.exit` calls" work are merged, the [original PR](https://github.com/conda/conda-build/pull/5255) will get rebased and will mostly focus on cleanup.